src/Knp/Menu/Matcher/Voter/RouteVoter.php

```diff
@@ -83,7 +83,7 @@ private function isMatchingRoute(array $testedRoute)
         $routeParameters = $this->request->attributes->get('_route_params', array());
 
         foreach ($testedRoute['parameters'] as $name => $value) {
-            if (!isset($routeParameters[$name]) || (string) $routeParameters[$name] !== (string) $value) {
+            if (!isset($routeParameters[$name]) || $routeParameters[$name] !== (string) $value) {
```

Comment:
Dumping the matcher gives `Symfony\Bundle\FrameworkBundle\Routing\Router`, and dumping the matcher inside this one gives `appDevUrlMatcher`

Apart from that :

```
  data_collector.router                                                Symfony\Bundle\FrameworkBundle\DataCollector\RouterDataCollector                                       
  fos_rest.router                                                      alias for "router"                                                                                     
  knp_menu.voter.router                                                Knp\Menu\Matcher\Voter\RouteVoter                                                                      
  monolog.logger.router                                                Symfony\Bridge\Monolog\Logger                                                                          
  router                                                               Symfony\Bundle\FrameworkBundle\Routing\Router                                                          
  router_listener                                                      Symfony\Component\HttpKernel\EventListener\RouterListener                                              
  templating.helper.router                                             Symfony\Bundle\FrameworkBundle\Templating\Helper\RouterHelper                                          
  web_profiler.controller.router                                       Symfony\Bundle\WebProfilerBundle\Controller\RouterController
```